lib/global_uid/server.rb

```diff
@@ -69,16 +69,21 @@ def allocate(klass, count: 1)
       #   Timeout.timeout is unpredictable
       Timeout.timeout(query_timeout, TimeoutException) do
         if count == 1
-          allocator.allocate_one(klass.global_uid_table)
+          allocator(klass).allocate_one
         else
-          allocator.allocate_many(klass.global_uid_table, count: count)
+          allocator(klass).allocate_many(count: count)
         end
       end
     end
 
     private
 
-    attr_accessor :connection_retry, :connection_timeout, :retry_at, :increment_by, :query_timeout, :allocator
+    attr_accessor :connection_retry, :connection_timeout, :retry_at, :increment_by, :query_timeout, :allocators
+
+    def allocator(klass)
+      table_name = klass.global_uid_table
```

Comment:
We probably want to memoize the `#global_uid_table` method now. It generates a couple of strings on each call.